clar2wasm/tests/oom-checker/main.rs

```diff
@@ -0,0 +1,116 @@
+#![cfg(test)]
+pub mod unit_tests;
+
+use clar2wasm::compile;
+use clar2wasm::datastore::Datastore;
+use clar2wasm::tools::{crosscheck, crosscheck_with_env, TestConfig, TestEnvironment};
+use clar2wasm::wasm_utils::get_type_in_memory_size;
+use clarity::types::StacksEpochId;
+use clarity::vm::costs::LimitedCostTracker;
+use clarity::vm::errors::{CheckErrors, Error};
+use clarity::vm::types::{
+    ListTypeData, QualifiedContractIdentifier, StandardPrincipalData, TypeSignature,
+};
+use clarity::vm::{ClarityVersion, Value};
+
+#[allow(clippy::expect_used)]
+pub fn as_oom_check_snippet(
+    snippet: &str,
+    args_types: &[TypeSignature],
+    epoch: StacksEpochId,
+    version: ClarityVersion,
+) -> String {
+    let compiled_module = Datastore::new()
+        .as_analysis_db()
+        .execute(|analysis_db| {
+            compile(
+                snippet,
+                &QualifiedContractIdentifier::new(
+                    StandardPrincipalData::transient(),
+                    ("foo").into(),
+                ),
+                LimitedCostTracker::new_free(),
+                version,
+                epoch,
+                analysis_db,
+            )
+            .map_err(|e| CheckErrors::Expects(format!("Compilation failure {e:?}")))
+        })
+        .expect("Could not compile snippet")
+        .module;
+
+    let memory_pages = compiled_module
+        .memories
+        .iter()
+        .next()
+        .expect("Couldn't find a memory")
+        .initial as usize;
+    let stack_pointer_value = match compiled_module
+        .globals
+        .iter()
+        .find(|g| g.name.as_ref().is_some_and(|name| name == "stack-pointer"))
+        .expect("Couldn't find stack-pointer global")
+        .kind
+    {
+        walrus::GlobalKind::Local(walrus::InitExpr::Value(walrus::ir::Value::I32(val))) => {
+            val as usize
+        }
+        _ => unreachable!("stack-pointer should be a locally declared global with a i32 value"),
+    };
+
+    let args_space_needed = args_types
+        .iter()
+        .map(|ty| get_type_in_memory_size(ty, false))
+        .sum::<i32>() as usize;
+
+    let free_space_on_memory_page = memory_pages * 65536 - stack_pointer_value;
+
+    format!(
+        "(define-constant ignore 0x{})\n{snippet}",
+        "00".repeat({
+            // we will need 8 bytes for the (offset, size) and 6 bytes for the name "ignore"
```

Comment:
make a const = 14 and use that here and below instead. (with the explanation)

Reply:
That's fair. I actually put the `ignore` name in a constant and use its length, I find it even cleaner.

Done in 5b1a8fbd